chia/wallet/wallet_blockchain.py

```diff
@@ -78,6 +80,8 @@ async def new_valid_weight_proof(self, weight_proof: WeightProof, records: List[
 
         self._sub_slot_iters = records[-1].sub_slot_iters
         self._difficulty = uint64(records[-1].weight - records[-2].weight)
+        await self._basic_store.set_object("SUB_SLOT_ITERS", self._sub_slot_iters)
```

Comment:
@altendky 
@altendky 
i want to make sure the 3 different writes to the basic_store are under the same transaction here do i just call them under 
        async with self.block_store.db_wrapper.writer():
?

Reply:
Yep, that's it.  It tracks connections and nesting etc internally.  The only caveat is that maybe someday we want to make the wrapper object private and so on, but for now, yup.